src/commandsAndMenu.tsx

```diff
@@ -480,18 +481,37 @@ export function addCommands(
               tooltip: trans.__('Refresh diff widget'),
               className: 'jp-git-diff-refresh'
             });
-            refreshButton.hide();
-            diffWidget.toolbar.addItem('refresh', refreshButton);
 
-            const refresh = () => {
-              refreshButton.show();
-            };
-
-            model.changed.connect(refresh);
-            widget.disposed.connect(() => {
-              model.changed.disconnect(refresh);
+            const resolveButton = new ToolbarButton({
+              label: trans.__('Mark as resolved'),
+              onClick: async () => {
+                try {
+                  const resolvedFile: string = await widget.getResolvedFile();
+                  // TODO update the file without the conflicts, then close the tab
+                  console.log(resolvedFile);
+                } catch (reason) {
+                  console.error(reason);
+                }
```

Comment:
so the resolved file stuff works now,

Turns out we created a resolve button in the PlainTextDiff, but forgot to remove it lol. Which caused our issues.

Whats the approach/functions for updating this file with the new contents, and removing the diff panel?


I assume the latter is just 
```suggestion
              onClick: async () => {
                try {
                  const resolvedFile: string = await widget.getResolvedFile();
                  // TODO update the file
                  widget.dispose();
                } catch (reason) {
                  console.error(reason);
                }
```

but not sure about the updating part